### PR TITLE
refactor(cmd): extract run() startup into staged init phases (#1391)

### LINKED ISSUE #1391: refactor(cmd): extract run() startup into staged init phases
[mode: plan]
[model: sonnet]
[effort: high]

## Problem
`cmd/stillwater/main.go:102 run()` is a monolithic 700-line startup function with cyclomatic complexity 61. It composes config load, logging, DB open, encryption keying, auth registry, library backfill, listener wiring, TLS/ACME/HTTP3 selection, watcher/maintenance/event-bus startup, and shutdown sequencing. No unit test reaches it; only integration boots exercise it.

## Strategy
Extract into a typed `Application` struct with phased init: `loadConfig`, `setupLogging`, `openStorage`, `wireSecurity`, `buildServices`, `startListeners`, `startBackgroundWorkers`, `runUntilSignal`. Each phase returns errors that bubble. Use functional options for testing seams (mock encryption key resolver, etc.).

## Acceptance criteria
- [ ] `run()` cyclomatic ≤10.
- [ ] Each phase has unit tests covering its happy path and error paths.
- [ ] TLS-source selection logic and encryption-key resolution become independently testable.
- [ ] Behavior unchanged (verify with manual UAT + smoke).

## Risk
Largest single PR in M49.5. Ships LAST per plan (W4) -- a botched startup-refactor blocks every subsequent UAT.

## Charter
M49.5 (Code Health and Hardening). Plan issue B1.

### PR DESCRIPTION
**Pull request by sydlexius**
## Summary

Decomposes the monolithic 656-line `run()` in `cmd/stillwater/main.go` (cyclomatic 64) into an `Application` struct with ordered init phases. `run()` is now cyclomatic 7; `buildServices` is cyclomatic 7 after sub-extraction. Closes #1391.

Behavior change: `resolveEncryptionKey` now returns an error when the generated key cannot be persisted (previously warn-and-continue). The old behavior silently caused the next startup to generate a different key, making every previously-encrypted secret unrecoverable. Operators with a read-only key directory will now see startup fail with a clear error instead of losing secrets on the next restart.

Preserves the inbound-first / outbound-second webhook drain order from #1514 and #1517 verbatim. Final M49.5 wave (W4.4A) -- completes the milestone.

## What changed

- **Phased init:** `loadConfig` -> `setupLogging` -> `openStorage` -> `wireSecurity` -> `buildServices` -> `startListeners`. Each phase has a precondition guard so out-of-order calls return an error at the phase boundary rather than nil-deref deep in production code.
- **Lifecycle defers** registered in `run()` at point-of-acquisition. LIFO order: scanner shutdown -> webhook inbound drain -> outbound drain -> listener stop -> `eventBus.Stop` -> `logManager.Close` -> `db.Close`. Cleanup now fires correctly on partial-init failure.
- **`buildServices` sub-extraction:** `wireAuth`, `wireProviders`, `wireRuleEngine`, `wireEventBus`, `wireInfraServices`, `wireEventSubscriptions`, `resolveRuleSchedule`.
- **Testing seams:** `WithEncKeyResolver` and `WithDBOpener` functional options on `newApplication` enable unit tests of `wireSecurity` and `openStorage` error paths without touching disk or env.
- **New tests** in `cmd/stillwater/phases_test.go` cover each phase's happy + error paths, `buildTLSStatus` branches, `resolveEncryptionKey` edge cases, `loadConfig` with a malformed file, and `getDBBoolSetting` false-value parsing.

## Test plan

- [x] go test -race ./... passes
- [x] scripts/pre-push-gate.sh passes (cyclomatic, patch coverage, fuzz matrix, lint, all green)
- [x] UAT: clean boot via scripts/dev-restart.sh; SIGTERM produces ordered shutdown log; /healthz returns 200
- [x] UAT: cleanup LIFO order verified (no closing-database error on clean exit -- db.Close returned nil)
- [x] Inbound + outbound webhook drain calls preserved at the documented LIFO position
- [ ] Reviewer: confirm the resolveEncryptionKey behavior change is acceptable

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Refactor**
  * Restructured startup into a staged Application lifecycle with clearer initialization, service wiring, and coordinated background worker startup.
  * Consolidated background workers (schedulers, scanners, session cleanup, filesystem watcher) with safer defaults and health bootstrap; TLS port logged only in split‑port mode.
  * Improved shutdown sequencing to drain inbound webhooks then outbound deliveries before stopping scanners.

* **Bug Fixes**
  * Startup now fails on failures creating or persisting encryption keys and when an existing key cannot be loaded.
  * DB-backed settings helpers now require an explicit context for queries.

* **Tests**
  * Added comprehensive end‑to‑end tests for config, logging, storage, security, and DB-backed settings.

[Review Change Stack](https://app.coderabbit.ai/change-stack/sydlexius/stillwater/pull/1531?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: cmd/stillwater/main.go [hunk 1]
```diff
@@ -779,20 +922,18 @@ func resolveEncryptionKey(cfg *config.Config, logger *slog.Logger) (string, erro
...
+		return "", fmt.Errorf("creating data directory for encryption key: %w", err)
 	}
 
 	if err := os.WriteFile(keyFile, []byte(key+"\n"), 0o600); err != nil {
-		logger.Warn("could not save encryption key to file",
-			slog.String("path", keyFile), slog.Any("error", err))
-	} else {
-		logger.Warn("generated new encryption key -- back up this file",
-			slog.String("path", keyFile))
+		return "", fmt.Errorf("saving encryption key to file %s: %w", keyFile, err)
```
[the remark this answers is not in the document]
Comment (sydlexius) [reply cmd/stillwater/main.go]: Fixed in 2d5f8761. Switched to filesystem.WriteFileAtomic so the encryption.key persist now uses the project's tmp/bak/rename pattern; a crash or disk-full mid-write can no longer leave a truncated key on disk.

File: cmd/stillwater/phases_test.go [hunk 2]
```diff
@@ -0,0 +1,493 @@
...
+}
+
+// --- loadConfig tests ---
+
+func TestLoadConfig_DefaultPath(t *testing.T) {
+	// Unset SW_CONFIG_PATH so that loadConfig falls back to the default
+	// path. The default (/config/config.toml) will not exist in CI, but
+	// config.Load is expected to succeed with defaults when the file is
+	// absent.
+	t.Setenv("SW_CONFIG_PATH", "")
```
[the remark this answers is not in the document]
Comment (sydlexius) [reply cmd/stillwater/phases_test.go]: Fixed in 2d5f8761. Replaced t.Setenv with an os.Unsetenv + t.Cleanup restore so the test now exercises the env-absent branch as documented.
